### ISSUE TITLE
yarn install --production does an engine check on devDependencies

### ISSUE THREAD

**Issue by tushar00jain** [yarnpkg/yarn#5796]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**
`yarn install --production` does an engine check on devDependencies

**What is the expected behavior?**
`yarn install --production` should not do an engine check on devDependencies

**Please mention your node.js, yarn and operating system version.**

**Comment by rally25rs** [comment]
Yarn still resolves all devDependencies to determine their hoisted location. This would be related to #3630

Even if that behavior doesn't change, it does seem reasonable that we could skip the engine check for dev deps during a `--production` install.

**Comment by tushar00jain** [comment]
Yes, it would be nice if the package wants to support older versions of node but use devDependencies that will not support the older versions

**Comment by EfstathiadisD** [comment]
I will give this a go if you do not mind.

**Comment by cavishek39** [comment]
Hi @rally25rs what is the status of the issue?
